src/main/java/java/lang/String.java

```diff
@@ -4266,8 +4266,8 @@ public static String valueOf(float f) {
      * actual program.
      */
     public static String valueOf(double d) {
-        // DIFFBLUE MODEL LIBRARY This is treated internally in CBMC
-        return CProver.nondetWithoutNullForNotModelled();
+        // DIFFBLUE MODEL LIBRARY we cast the number down to float because string solver only knows how to convert floats to string
```

Comment:
:hammer_and_pick: Could you wrap this line at 80 characters please?
:question: Also, is it a new thing that the string solver only converts floats and not doubles? If not, how did this work before?

Reply:
It's not a new thing, before we'd either crash completely or not crash and then return a conversion which was likely to be incorrect.